### PR TITLE
Fixes #10453 - API for starting a deployment

### PR DESCRIPTION
**Pull request by tstrachota**

### CODE REVIEW

**Comment by tstrachota** [comment]
@ShimShtein rebased

File: test/functional/api/v2/deployments_controller_test.rb [hunk 1]
```diff
@@ -2,17 +2,43 @@
...
+        'Task2: !task:FakeTask'
+      ].join("\n")
+    )
+    @deployment_params = { :name => 'A Deployment', :stack_id => @stack.id }
+    @deployment = ForemanDeployments::Deployment.create(
+      :name => 'Another deployment',
+      :configuration => ForemanDeployments::Configuration.new(:stack => @stack)
+    )
+
+    ForemanDeployments.registry.register_task(FakeTask)
```
Comment (ShimShtein) [inline test/functional/api/v2/deployments_controller_test.rb]: Same as in #13, please avoid changing global setting in a single test. It's better to isolate the setting.

Comment (tstrachota) [inline test/functional/api/v2/deployments_controller_test.rb]: Correct, I used that pattern on multiple places. I need to fix it everywhere.

**Comment by tstrachota** [comment]
@ShimShtein updated

**Comment by ShimShtein** [comment]
@tstrachota, this PR needs rebasing, since ValidationResult has moved in #15 (it was merged first)

**Comment by tstrachota** [comment]
@ShimShtein rebased, thanks for letting me know

**Comment by ShimShtein** [comment]
Merged as https://github.com/theforeman/foreman_deployments/commit/619bc9fcb66e0e03effed1b0e565f90f70b04e72

Thanks @tstrachota

**Pull request closed without merging**